automations/js/src/project_thread_updates.js

```diff
@@ -61,10 +65,10 @@ module.exports = async ({ github, core }) => {
     const isDryRun = process.env.DRY_RUN === 'true' ?? false
 
     const currentDate = new Date()
-    const fourteenDaysAgo = new Date(
+    const requiredUpdatedByDate = new Date(
       currentDate.getFullYear(),
       currentDate.getMonth(),
-      currentDate.getDate() - 14
+      currentDate.getDate() - DAYS_UPDATED_WITHIN
     )
```

Comment:
This won't work if the current day of the month is less than 14:

```
> new Date(2023, 6, 9 - 14)
2023-06-24T14:00:00.000Z
```

It rolls over into the future (quite unexpectedly).

It needs some logic along these lines:

```js
let month, day
if (currentDate.getDate() < DAYS_UPDATED_WITHIN) {
  month = currentDate.getMonth() - 1
  previousMonth = new Date(currentDate.getYear(), month) // defaults to last day of month
  day = 
    previousMonth.getDate() // last day of month
    - (DAYS_UPDATED_WITHIN - currentDate.getDate()) // days overlapping with previous month
}
// etc...
```

But that also needs to include some contingency for overlapping years and needs to handle January as a special case.

https://github.com/moment/luxon/ may be a good option here to simplify things. However, it would need to be installed in the environment of the action, [docs here for that](https://github.com/actions/github-script#use-npm-packages). Adding the year might be the last edge case but I haven't thought about it long enough to be sure :slightly_smiling_face:

Reply:
Oh gosh, I can't believe I merged this 🤦 the simplest fix without introducing dependencies is just to convert to milliseconds:

`const requiredUpdatedByDate = new Date(currentDate.getTime() - (DAYS_UPDATED_WITHIN * 24 * 60 * 60 * 1000));`

Reply:
Addressed in 1f2fa4c92.

Reply:
Ah, of course! That is a much better solution.